reposilite-backend/src/test/kotlin/com/reposilite/auth/LdapAuthenticatorTest.kt

```diff
@@ -18,123 +18,191 @@ package com.reposilite.auth
+                    host = "host",
+                    name = token.name,
+                    secret = secret
+                )
             )
-        )
 
-        val accessToken = assertOk(authenticationResult)
-        assertThat(accessToken.name).isEqualTo("Bella Swan")
-        assertThat(accessToken).isEqualTo(accessTokenFacade.getAccessToken("Bella Swan"))
+            val accessToken = assertOk(authenticationResult)
+            assertThat(accessToken.name).isEqualTo("Bella Swan")
+            assertThat(accessToken).isEqualTo(token)
+        }
+
     }
 
-    @Test
-    fun `should authenticate existing ldap user`() {
-        val (token, secret) = accessTokenFacade.createAccessToken(
-            CreateAccessTokenRequest(
-                type = TEMPORARY,
-                name = "Bella Swan",
-                secret = "secret"
+    @Nested
+    inner class GH2054 {
+
+        init {
+            ldapConfiguration.update {
+                it.copy(
+                    baseDn = "ou=people,dc=domain,dc=com",
+                    searchUserDn = "cn=readonly,dc=domain,dc=com",
+                    searchUserPassword = "search-secret",
+                    typeAttribute = "posixAccount",
+                    userAttribute = "uid",
+                    userFilter = "(memberOf=cn=users,ou=reposilite,dc=domain,dc=com)",
+                    userType = it.userType
+                )
+            }
+
+            createLdapServer(
+                ldapSettings = ldapConfiguration.get(),
+                baseDns = listOf("dc=domain,dc=com", "ou=people,dc=domain,dc=com"),
+                credentials = mapOf(
+                    "cn=readonly,dc=domain,dc=com" to "search-secret",
+                    "uid=mykola,ou=people,dc=domain,dc=com" to "mykola-secret"
+                )
             )
-        )
+        }
+
+        @BeforeEach
+        fun setup() {
+            ldapServer.add("dn: dc=domain,dc=com", "objectClass: top", "objectClass: domain")
+
+            ldapServer.add("dn: ou=people,dc=domain,dc=com", "objectClass: organizationalUnit", "objectClass: top")
+            ldapServer.add("dn: ou=reposilite,dc=domain,dc=com", "objectClass: organizationalUnit", "objectClass: top")
 
-        val authenticationResult = authenticator.authenticate(
-            Credentials(
-                host = "host",
-                name = token.name,
-                secret = secret
+            ldapServer.add("dn: cn=users,dc=domain,dc=com", "ou: people", "objectClass: posixAccount", "memberOf: ou=people,dc=domain,dc=com")
+            ldapServer.add("dn: cn=readonly,dc=domain,dc=com", "ou: reposilite", "uid: readonly", "objectClass: posixAccount", "memberOf: cn=users,ou=reposilite,dc=domain,dc=com")
+
+            ldapServer.add("dn: uid=mykola,ou=people,dc=domain,dc=com", "ou: reposilite", "objectClass: posixAccount", "memberOf: cn=users,ou=reposilite,dc=domain,dc=com")
+        }
+
+        @Test
+        fun `should authenticate mykola`() {
```

Comment:
@dzikoysk Please correct me if I'm wrong but understand this test like only "mykola" is allowed to login. But as I described it in the ticket, only "readonly" should be allowed to connect LDAP and all other should be blocked. This is how BindDN works.

Reply:
Okay, so that's not how I understood the issue. From the logs you've provided I thought you can successfully login with readonly (conn=1021), but you cannot do this with mykola (conn=1022). Could you specify your LDAP server config that I can use in this test to go through your issue? I feel like it can be done in multiple ways and I'm currently trying to guess what setup you actually have :thinking:

Reply:
[ldap.txt](https://github.com/dzikoysk/reposilite/files/14311320/ldap.txt)

Reply:
[ldap.txt](https://github.com/dzikoysk/reposilite/files/14311332/ldap.txt)

Reply:
> Okay, so that's not how I understood the issue. From the logs you've provided I thought you can successfully login with readonly (conn=1021), but you cannot do this with mykola (conn=1022).

Yes, you understand it correct. User "mykola" shouldn't have expected to LDAP server. Access is granted only for user specified in Reposilite configuration and none other.